components/Launchpad/LaunchpadStats.js

```diff
@@ -0,0 +1,132 @@
+import { useEffect, useState } from 'react'
+import ReactTooltip from 'react-tooltip'
+import { projectStatus } from './LaunchpadItem'
+import LaunchpadStatsLoader from './LaunchpadStatsLoader'
+import TimeLaunchpad from './TimeLaunchpad'
+
+const LaunchpadStats = ({ project, isEnded, isEndedComing, isValidating }) => {
+	const [showTooltip, setShowTooltip] = useState(false)
+	const [mintDuration, setMintDuration] = useState()
+	const [isEndedUpcoming, setIsEndedUpcoming] = useState(false)
+	const randomID = 'launchpad-stats'
+
+	useEffect(() => {
+		setShowTooltip(true)
+		getMintDuration()
+	}, [project])
+
+	const getMintDuration = () => {
+		const startDate = new Date(project?.mint_details[0].started_at)
+		const endedDate = new Date(project?.mint_details[0].ended_at)
+		const diff = endedDate.getTime() - startDate.getTime()
+		setMintDuration(Math.floor(diff / 1000 / 60 / 60))
+	}
```

Comment:
`mintDuration` kira2 perlu masuk useEffect dan state ga yah? soalnya best practicenya [gini](https://beta.reactjs.org/learn/you-might-not-need-an-effect#how-to-remove-unnecessary-effects)
```js
const getMintDuration = (startDateTime, endDateTime) => {
  const startDate = new Date(startDateTime)
  const endedDate = new Date(endDateTime)
  const diff = endedDate.getTime() - startDate.getTime()
  setMintDuration(Math.floor(diff / 1000 / 60 / 60))
}

const LaunchpadStats = ({ project, isEnded, isEndedComing, isValidating }) => {
	const mintDuration = getMintDuration(project?.mint_details[0].started_at, project?.mint_details[0].ended_at)
  
  // more code
}
```

Reply:
mantap, noted hu